sdk/core/Azure.Core/api/Azure.Core.netstandard2.0.cs

```diff
@@ -516,6 +516,17 @@ public partial class BearerTokenAuthenticationPolicy : Azure.Core.Pipeline.HttpP
         public override void Process(Azure.Core.HttpMessage message, System.ReadOnlyMemory<Azure.Core.Pipeline.HttpPipelinePolicy> pipeline) { }
         public override System.Threading.Tasks.ValueTask ProcessAsync(Azure.Core.HttpMessage message, System.ReadOnlyMemory<Azure.Core.Pipeline.HttpPipelinePolicy> pipeline) { throw null; }
     }
+    public partial class BearerTokenChallengeAuthenticationPolicy : Azure.Core.Pipeline.HttpPipelinePolicy
+    {
+        public BearerTokenChallengeAuthenticationPolicy(Azure.Core.TokenCredential credential, System.Collections.Generic.IEnumerable<string> scopes) { }
+        public BearerTokenChallengeAuthenticationPolicy(Azure.Core.TokenCredential credential, string scope) { }
+        protected System.Collections.ObjectModel.ReadOnlyCollection<string> Scopes { get { throw null; } }
+        protected virtual System.Threading.Tasks.ValueTask AuthorizeRequestAsync(Azure.Core.HttpMessage message, bool async) { throw null; }
+        protected virtual System.Threading.Tasks.ValueTask<bool> AuthorizeRequestOnChallengeAsync(Azure.Core.HttpMessage message, bool async) { throw null; }
```

Comment:
We shouldn't expose the `bool async` pattern publicly.

Reply:
Do you have an alternative in mind? This needs to be at least protected so deriving classes can override it, and it needs to be asynchronous because it likely will need to acquire a token which could involve a network call.

Reply:
``` C#
        protected virtual System.Threading.Tasks.ValueTask<bool> AuthorizeRequestOnChallengeAsync(Azure.Core.HttpMessage message) { throw null; }
        protected virtual bool AuthorizeRequestOnChallenge(Azure.Core.HttpMessage message) { throw null; }
```
?

Reply:
I see. It's a bit inconvenient to implementers who will essentially always have to implement both, and will probably use the `bool async` pattern underneath. But if we don't expose this pattern anywhere else already, I'm fine with making implementers write a few more lines, since most likely only client libraries will be extending it.